### PR TITLE
.circleci: Add CUDA 11 to nightly binary builds

### PR DESCRIPTION
**Pull request by seemethere**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #43370 DO NOT MERGE, testing cuda 11
* **#43366 .circleci: Add CUDA 11 to nightly binary builds**

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D23348556](https://our.internmc.facebook.com/intern/diff/D23348556)

### CODE REVIEW

File: .circleci/cimodel/data/binary_build_data.py [hunk 1]
```diff
@@ -51,13 +51,16 @@ def get_processor_arch_name(cuda_version):
...
     # Skip CUDA-9.2 builds on Windows
-    windows=([v for v in dimensions.CUDA_VERSIONS if v != '92'], OrderedDict(
-        wheel=dimensions.STANDARD_PYTHON_VERSIONS,
-        conda=dimensions.STANDARD_PYTHON_VERSIONS,
-        libtorch=[
-            "3.7",
-        ],
-    )),
+    windows=(
+        [v for v in dimensions.CUDA_VERSIONS if v not in ['92', '110']],
```
Comment (seemethere) [inline .circleci/cimodel/data/binary_build_data.py:55]: Disabling CUDA 11 builds here for windows temporarily to unblock linux builds

**Review by malfet — APPROVED** [review]

**Comment by seemethere** [comment]
This has passed as of: https://app.circleci.com/pipelines/github/pytorch/pytorch/203448/workflows/0385ac58-9339-4a73-99aa-948c00d732d6

File: .circleci/cimodel/data/binary_build_data.py (hunk 1 above)
Comment (peterjc123) [reply .circleci/cimodel/data/binary_build_data.py:55]: Hey, please cc me or the windows team about it next time.

File: .circleci/scripts/binary_linux_test.sh [hunk 2]
```diff
@@ -40,7 +40,7 @@ if [[ "$PACKAGE_TYPE" == conda ]]; then
     else
       cu_ver="${DESIRED_CUDA:2:2}.${DESIRED_CUDA:4}"
     fi
-    retry conda install -yq -c pytorch "cudatoolkit=\${cu_ver}"
+    retry conda install -yq -c nvidia -c pytorch "cudatoolkit=\${cu_ver}"
```
Comment (peterjc123) [inline .circleci/scripts/binary_linux_test.sh:43]: This change is not needed now since the `cudatoolkit` package is available in the `defaults` channel.

**Pull request closed without merging**